paper-server/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java

```diff
@@ -2122,6 +2128,69 @@ public static boolean callEntityInteractEvent(Entity nmsEntity, Block block) {
         return !event.isCancelled();
     }
 
+    public static List<BlockPos> handleExplodeEvent(ServerExplosion serverExplosion, List<BlockPos> blockPositions) {
+        // First convert the blockPositionList related to explosion to Bukkit
+        World bworld = serverExplosion.level().getWorld();
+
+        List<Block> blockList = new ObjectArrayList<>();
+        for (int posBlockPositionList = blockPositions.size() - 1; posBlockPositionList >= 0; posBlockPositionList--) {
```

Comment:
Since you already reverse the order for the old events, redoing it will make the list inverted compared to the old events. The block list should be consistent across events and ideally the list should remains intact after the events (so not inverted)

Reply:
Then just convert in all cases and not mess with order?

Reply:
I just checked why they reverse the order like that, and it's a relic of the past back in 1.2 which doesn't make sense anymore in modern version.
However i'm tired of this PR, so i will take over.

Reply:
Okay... Sorry for the issues and thanks for the help...